sdk/cmake/host-tools.cmake

```diff
@@ -15,11 +15,30 @@ function(setup_host_tools)
         set(HOST_EXE_SUFFIX ".exe")
     endif()
 
+    if (CMAKE_SYSTEM_PROCESSOR EQUAL CMAKE_HOST_SYSTEM_PROCESSOR)
+        set(HOST_TOOLS_CMAKE_COMMAND "${CMAKE_COMMAND}")
+        message("Not cross-compiling, no special host-tools cmake command")
+    elseif (MSVC)
+        message("Compiling on ${CMAKE_HOST_SYSTEM_PROCESSOR} for ${CMAKE_SYSTEM_PROCESSOR} (MSVC)")
+        if(MSVC_VERSION EQUAL 1900)
+            set(HOST_TOOLS_CMAKE_COMMAND "$ENV{VSINSTALLDIR}\\VC\\vcvarsall.bat" $ENV{PROCESSOR_ARCHITECTURE} && "${CMAKE_COMMAND}")
+        elseif(MSVC_VERSION GREATER_EQUAL 1910)
+            # 2017 and 2019 use the same folder structure
+           set(HOST_TOOLS_CMAKE_COMMAND "$ENV{VSINSTALLDIR}\\VC\\Auxiliary\\Build\\vcvarsall.bat" $ENV{PROCESSOR_ARCHITECTURE} && "${CMAKE_COMMAND}")
+        else()
+            message(FATAL "Unable to figure out vcvarsall path")
+        endif()
+    else()
+    set(HOST_TOOLS_CMAKE_COMMAND "${CMAKE_COMMAND}")
+    message("Cross-compiling on non-msvc, no special host-tools cmake command")
+endif()
+
     ExternalProject_Add(host-tools
         SOURCE_DIR ${REACTOS_SOURCE_DIR}
         PREFIX ${REACTOS_BINARY_DIR}/host-tools
         BINARY_DIR ${REACTOS_BINARY_DIR}/host-tools/bin
-        CMAKE_ARGS -UCMAKE_TOOLCHAIN_FILE -DARCH:STRING=${ARCH} -DCMAKE_INSTALL_PREFIX=${REACTOS_BINARY_DIR}/host-tools -DTOOLS_FOLDER=${REACTOS_BINARY_DIR}/host-tools/bin
+        CMAKE_COMMAND ${HOST_TOOLS_CMAKE_COMMAND}
+        CMAKE_ARGS -UCMAKE_TOOLCHAIN_FILE -DCMAKE_INSTALL_PREFIX=${REACTOS_BINARY_DIR}/host-tools -DTOOLS_FOLDER=${REACTOS_BINARY_DIR}/host-tools/bin
```

Comment:
I removed ` -DARCH:STRING=${ARCH}` here since the host-tools do not need to be built for our target arch, is that correct @zefklop ?

Reply:
No it's not. dbghelp_host relies on that See [here](https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/dbghelp/CMakeLists.txt;hb=e044df87366d52fa5dfce50e994e3d8e2a2965c7#l21)

Reply:
ok, thanks.